src/components/admin/form-components/SearchBar.jsx

```diff
@@ -97,23 +127,25 @@ export class SearchBar extends React.Component {
 
 SearchBar.propTypes = {
   handleClick: PropTypes.func.isRequired,
-  suggestions: PropTypes.arrayOf(
-    PropTypes.shape({
-      title: PropTypes.string.isRequired,
-      slug: PropTypes.string,
-      id: PropTypes.number.isRequired,
-    }),
-  ),
-  updateSuggestions: PropTypes.func.isRequired,
+  getSuggestions: PropTypes.func.isRequired,
+  mode: PropTypes.string.isRequired,
   fullWidth: PropTypes.bool,
   disabled: PropTypes.bool,
   enableAdd: PropTypes.bool,
-  category: PropTypes.string.isRequired,
+  slugify(props, propName) {
+    if (
+      props.enableAdd === true &&
+      (props[propName] === undefined || typeof props[propName] !== 'function')
+    ) {
+      return new Error('Please provide a slugify function!');
+    }
+    return null;
+  },
```

Comment:
Do you happen to know or have tested whether putting `.isRequired` behind this and removing your check whether it's undefined would work? That would probably also stop the linter from complaining about defaultProps right? If it doesn't no problem, then I guess it's just a necessary evil

Reply:
I haven't. But I did is this way, because I only want to require it if enableAdd is set.

Reply:
Ahh yes, I must've gotten confused, absolutely right, and in that case I guess we also want there to be a default value anyway